assets/global.js

```diff
@@ -6,6 +6,21 @@ function getFocusableElements(container) {
   );
 }
 
+document.querySelectorAll('[id^="Details-"] summary').forEach((summary) => {
```

Comment:
> Any reason we couldn't just target `summary` elements in general?

Not all instances of `details`/`summary` convey the same meaning. When JS is available, it's used to enhance the user experience by providing greater AT support and more familiar navigational patterns (support for native `details`/`summary` can be inconsistent.) Some instances convey a disclosure pattern, where others instances convey a modal context.

Ideally there'd be only two sources of truth; one for each type of pattern. I'm not overly opinionated on how we target either in the DOM, but it should be obvious to anyone reading the code. And of course, any `id` that's produced must be unique, otherwise there's the possibility of a target mismatch as well as invalid HTML.

Reply:
I think IDs was the chosen way because we wanted to stir away from using data-attributes too often like we did in previous themes.
 I was trying to see if I could move everything to the `details-disclosure` file. But there are a few side effects. Messes with the styling for the menu drawer, functionality of the facet filtering and also collapsible tabs on the product page. 

So instead I removed the `keyup` listener from `DetailsDisclosure` since it's already in that other function. 

Also I noticed that when you clicked outside of the summary content when it's open, it closes it but doesn't remove `aria-hidden="true"` unless you click on the summary element itself to close it or press the ESC key.

Let me know what you both think 👍

Reply:
Yeah I've added something to the `close()` function so it toggles the attribute 👌